client/landing/stepper/declarative-flow/internals/steps-repository/woo-transfer/index.tsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable no-console */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from 'react';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import { transferStates } from 'calypso/state/atomic/transfers/constants';
+import { ONBOARD_STORE, SITE_STORE } from '../../../../stores';
+import type { Step } from '../../types';
+
+const wait = ( ms: number ) => new Promise( ( res ) => setTimeout( res, ms ) );
+
+const WooTransfer: Step = function WooTransfer( { navigation } ) {
+	const { submit } = navigation;
+	const { setPendingAction, setProgress } = useDispatch( ONBOARD_STORE );
+	const {
+		requestAtomicSoftwareStatus,
+		requestLatestAtomicTransfer,
+		initiateAtomicTransfer,
+	} = useDispatch( SITE_STORE );
+	const site = useSite();
+
+	const siteId = site?.ID;
+
+	const {
+		getSiteLatestAtomicTransfer,
+		getSiteLatestAtomicTransferError,
+		getAtomicSoftwareStatus,
+		// getAtomicSoftwareError,
+	} = useSelect( ( select ) => select( SITE_STORE ) );
+
+	useEffect( () => {
+		if ( ! siteId ) return;
+
+		setPendingAction( async () => {
+			const startTime = new Date().getTime();
+			const totalTimeout = 1000 * 180;
+			const maxFinishTime = startTime + totalTimeout;
+
+			// Initiate transfer
+			await initiateAtomicTransfer( siteId, 'woo-on-plans' );
+
+			// Poll for transfer status
+			let stopPollingTransfer = false;
+
+			while ( ! stopPollingTransfer ) {
+				await wait( 3000 );
```

Comment:
Maybe! The idea behind it is that it would wait the first 3s after requesting the transfer status. because it wouldn't be ready so fast.

Reply:
You are right, transfer takes a relatively long time

Reply:
We could change the loop for a setInterval, but then, instead of just using an `async` function to return a promise that is resolved in the end of the logic, we would have to return a `new Promise()`, use the intervals, and then resolve if manually later. Since there are 2 different lopping, we would also have to add extra logic so the second part only starts after the first.